libs/blocks/merch-cards/merch-cards.js

```diff
@@ -0,0 +1,257 @@
+import '../../deps/merch-cards.js';
+import '../../deps/merch-card.js';
+import '../../deps/commerce.js';
+import '../merch/merch.js';
+import '../merch-card/merch-card.js';
+import { createTag, decorateLinks, getConfig, loadBlock, loadStyle } from '../../utils/utils.js';
+import { replaceText } from '../../features/placeholders.js';
+
+const DIGITS_ONLY = /^\d+$/;
+
+const LITERAL_SLOTS = [
+  'searchText',
+  'filtersText',
+  'sortText',
+  'popularityText',
+  'alphabeticallyText',
+  'noResultText',
+  'resultText',
+  'resultsText',
+  'searchResultText',
+  'searchResultsText',
+  'searchResultMobileText',
+  'searchResultsMobileText',
+  'noSearchResultsText',
+  'noSearchResultsMobileText',
+  'showMoreText',
+];
+
+// allows improve TBT by returning control to the main thread.
+// eslint-disable-next-line no-promise-executor-return
+const makePause = async () => new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
Can you share some data around this TBT optimization? It'd be interesting to understand the impact of interrupting processing in the `initMerchCards` function further down.

Reply:
i think it would deserve some central discussion (and feature!) as it as been used by @narcis-radu and @mokimo iirc on feds

Reply:
(not saying we need feature/discussion before end of that PR)

Reply:
> i think it would deserve some central discussion (and feature!) as it as been used by @narcis-radu and @mokimo iirc on feds

Right, a TBT optimization discussion sounds like a good idea 👍

Reply:
@honstar we have around 93 cards(fragment, block) in the catalog. We don't need to initialise all of them at the same time to render. It is increasing TBT considerably.
Here by interrupting the processes and giving the UI thread chance to do something else, we keep the TBT under control.

Reply:
the thing is, we don't know which card is the first card, and the card metadata must be initialised so that we know whether it must be displayed or not.
So we should still eagerly initialise them as early as possible, but with mini fake pauses :)